posydon/utils/common_functions.py

```diff
@@ -218,26 +218,40 @@ def roche_lobe_radius(m1, m2, a_orb=1):
     """
     
     ## catching if a_orb is an empty array or is an array with invalid separation values
-    if isinstance(a_orb, np.ndarray) and (not np.any(a_orb) or np.any(a_orb <=0)):
-        Pwarn("Trying to compute RL radius for binary with no separation", "EvolutionWarning")
-        a_orb = np.full_like(a_orb, np.nan)
+    if isinstance(a_orb, np.ndarray):
+        ## if array is empty (or all zeros), fill with NaN values
+        if a_orb.size == 0:
+            Pwarn("Trying to compute RL radius for binary with invalid separation", "EvolutionWarning")
+            a_orb = np.full_like(a_orb, np.nan, dtype=np.float64)
+        ## if array contains invalid values, replace with NaN 
+        elif np.any(a_orb < 0):
```

Comment:
Here we should use `a_orb <= 0`. The same two lines later.
(Alternatively we should use <0 for floats as well, in case we'd like to allow for a 0 separation. But I think we don't want that.)